honeybee_energy/properties.py

```diff
@@ -25,19 +25,17 @@ def construction(self):
         construction-set when the face is added to a zone. For a free floating face the
         construction will be None unless it is set by user.
         """
-        if self._construction:
-            # set by user
+        if self._construction:  # set by user
             return self._construction
-        elif self._constructionset:
-            # set by parent zone
+        elif self._constructionset:  # set by parent zone
             return self._constructionset(self._face_type, self._boundary_condition)
         else:
             return None
 
     @construction.setter
     def construction(self, value):
         if value:
-            assert isinstance(value, Construction), \
+            assert isinstance(value, OpaqueConstruction), \
```

Comment:
I know this is just an update to the existing code but you probably want to do this check against the base construction class and not only OpaqueConstruction.

Reply:
I tried to understand what your plan is in `honeybee-core` for aperture faces but, from what I can tell, they are not there now.  Once you add apertures, I can write this correctly for both opaque and glazing surfaces instead of assuming everything is opaque.
Whatever the case, you should not be able to assign a window construction to an opaque parent surface and vice-versa.  This is something we should catch when it is assigned rather than waiting for OpenStudio / E+ to give us an error.

Reply:
> Whatever the case, you should not be able to assign a window construction to an opaque parent surface and vice-versa. This is something we should catch when it is assigned rather than waiting for OpenStudio / E+ to give us an error.

This is true and shows a limitation of the current design. Properties class is currently used for all different face types. We might need to subclass Properties class then and assign different ones to different Face types and see how that will work out. I'm going to close this for now.